Learning Paths (#794)
temporalio/documentation: docs/learning-paths/background-checks/project-narrative.md

---
id: project-narrative
title: Welcome to the background-checks project
description: TODO
sidebar_label: Introduction
---

## What is the goal of this project?

> The goal of this project is to teach you, the developer, how to think about building Temporal Applications that have Long Running Human Driven Workflows using a Temporal SDK, by leading you through a comprehensive implementation within the context of a real-life use case.

Author: Not too late to change that.
But then wouldn't it be "Long-running Human-driven Workflow"?
Where the  first letter of the second word in the hyphenation is lowercase?

> The goal of this project is to teach you, the developer, how to think about building Temporal Applications that have Long Running Human Driven Workflows using a Temporal SDK, by leading you through a comprehensive implementation within the context of a real-life use case.

Reviewer: Welllll...you can get lots of debate on that issue. Both approaches are valid, but _Chicago_ 17th recommends capping the second word too. (I'm also inclined to add a comma following "-running", but that's a separate discussion.)

More specifically, _Chicago_ sez:

1. Always capitalize the first element. 
2. Capitalize any subsequent elements unless they are articles, prepositions, coordinating conjunctions (_and_, _but_, _for_, _or_, _nor_) , or such modifiers as _flat_ or _sharp_ following musical key symbols. 
3. If the first element is merely a prefix or combining form that could not stand by itself as a word (_anti_, _pre_, etc.), do not capitalize the second element unless it is a proper noun or proper adjective.

> The goal of this project is to teach you, the developer, how to think about building Temporal Applications that have Long Running Human Driven Workflows using a Temporal SDK, by leading you through a comprehensive implementation within the context of a real-life use case.

Reviewer: Having said all that, we're claiming a specific, Temporal-centric meaning for the term, so we can define it the way we want.

> The goal of this project is to teach you, the developer, how to think about building Temporal Applications that have Long Running Human Driven Workflows using a Temporal SDK, by leading you through a comprehensive implementation within the context of a real-life use case.

Author: Sounds like we have decided on:
> Human-Driven Long-Running Workflow

### How did Checkr solve these problems before Temporal?

To address those problems Checkr implemented a complex state machine using stand alone databases, Kafka queues, and micro-services.

But this approach introduced its own set of problems.

- Testing and applying updates and features to the live system was very hard, and there were no official means by which to do it.
- New engineers needed to understand the entire Checkr specific state machine architecture, regardless of the team or area they were working on.
- Even when things went as expected, the percentage of issues and failures at that scale meant that it was not reliable as desired.

Checkr understood that something needed to change to their approach and started looking for solutions.
In their search, they came across the Temporal Platform.

### Why is the Temporal Platform a great fit for this use case?

> Code based workflows appeal to developers.
> Business processes (Background Checks) map directly to Workflow functions.
> This results in a very close mapping of business requirements to code and vice versa.

Author: ```suggestion
Code based workflows appeal to developers as business processes (Background Checks) map directly to Workflow functions.
This results in a very close mapping of business requirements to code.
```
